.github/workflows/create-test-matrices/action.yml

```diff
@@ -0,0 +1,71 @@
+inputs:
+    language-name:
+        description: "Language name"
+        required: true
+        type: choice
+        options:
+            - java
+            - node
+            - python
+            - go
+            - C#
+    run-full-matrix:
+        description: "Run the full matrix"
+        required: true
+        type: boolean
+
+outputs:
+    engine-matrix-output:
+        description: "Engine matrix"
+        value: ${{ steps.load-engine-matrix.outputs.engine-matrix }}
+    host-matrix-output:
+        description: "Host matrix"
+        value: ${{ steps.load-host-matrix.outputs.host-matrix }}
+    version-matrix-output:
+        description: "Version matrix"
+        value: ${{ steps.create-lang-version-matrix.outputs.version-matrix }}
+
+runs:
```

Comment:
Should we echo each step? I think it will be too verbose, let's echo just helpful stuff for the dev which have CI fails on his PR or for the other triggers and what helpful for CI devs we leave as comments.
Then we can make them more specific for each need.

Reply:
Same for all matrices